Fix small issues

emmafgrimes opened the pull request:
## Describe the big picture - why are we doing this?
This PR fixes many small issues including:
- PyTorch Lightning Model Summary not working with latest version in CollieMinimalTrainer
- Upgrade torch to 1.10
- Before calling the model, check if the index is in-bound for `get_item_predictions` and `item_item_similarity`
- Remove default num_workers for Interactions DataLoaders
- Better catch the common ValueError that happens in in stratified splits, and add an option to `force_split` when users have only a single interaction
- Change Callable type hints to reflect the actual syntax for these (e.g. Callable[[str, str], int])
- Add in `get_user_predictions` and `get_user_embeddings` to BasePipeline and all model classes

## Any additional details to clarify code in the PR?

## How is this tested?
The tests were run on an EC2 instance and all passed. The tutorial notebooks were re-run.

A model was also fit using `pytorch_lightning==1.4.9` to test backwards compatibility.

## Pull Request Checklist
 - [x] Pull request includes a description of why we are doing this
 - [x] ``CHANGELOG`` has been updated
 - [x] Version in ``_version.py`` has been updated
 - [x] All tests in the ``tests`` folder pass with a local build
 - [ ] ``README`` has been updated (if applicable)
 - [ ] Documentation in ``docs`` has been updated (if applicable)
 - [x] ``requirements.txt`` and ``requirements-dev.txt`` have been recompiled (if applicable)
 - [x] Docker image can be built using ``docker build -t collie .``

## Screenshots or GIFs?
My first collie and open source contribution!
![](https://i.pinimg.com/originals/85/c1/95/85c195a2b3a90b217e664a734a447c5e.gif)

nathancooperjones commented:
Hi ShopRunner DS team! 

I went through and reviewed this PR over at our fork [here](https://github.com/the-mom-project/collie/pull/1). All the comments and back-and-forth can be viewed at the link above, so the version that is submitted here has already been thoroughly reviewed by me and is good to merge. 

We specifically made sure that the code is backwards compatible with previous versions of Collie so existing jobs at ShopRunner do not break, but just to be safe, you may choose to pin the version of Collie used in `k9academy` to be `1.1.2` just in case I missed something in my review. 

To merge it in, we just need another approval from someone on the SR team. If you want to do a full review, that is okay with us, or if you just want to do a light-scan and approve it, that also works. Whatever is easiest for you all! 

@gsganden @mcundiff3

gsganden commented:
This is awesome. Thank you @emmafgrimes and @nathancooperjones! I'll run it by the team to make sure we're on the same page about how to handle it and get back to you soon.

mcundiff3 commented on collie/model/base/base_pipeline.py:
```diff
@@ -722,6 +848,13 @@ def _get_item_embeddings(self) -> torch.tensor:
+        """``_get_user_embeddings`` should be implemented in all subclasses."""
+        raise NotImplementedError(
+            '``BasePipeline`` is meant to be inherited from, not used. '
+            '``_get_user_embeddings`` is not implemented in this subclass.'
```
I like this functionality! Very cool

mcundiff3 reviewed (changes requested):
This is AMAZING Emma, I'm so excited to have these changes in the repo. Thank you for contributing to the project.

I left a suggested change that we chatted about but should be a quick fix.

emmafgrimes commented:
@mcundiff3 thanks for the review and suggestion! I implemented the additional docstring where suitable, fixed a minor `**` typo, and updated testing `stratified_split` with a pytest parameter. Back to you!

mcundiff3 reviewed (approved):
Yes I'm so excited about this! Congrats on your first open source contribution. This is amazing!

I'll be your Gary and celebrate this:
![200](https://user-images.githubusercontent.com/26612213/150031756-ab1c4004-a93c-4ffc-949a-69513821272b.gif)
